blog/posts/2021-02-21-serverless-functions-ember-glimmer.md

```diff
@@ -0,0 +1,212 @@
+title: Serverless Functions in Ember and Glimmer
+---
+
+I have recently spent some time looking at [API Routes](https://nextjs.org/docs/api-routes/introduction) in Next.js and I must say that I've been very happy with the value-add and developer experience / ergonomics they provide. 
+
+I really wanted to enjoy that experience in Ember and Glimmer and that is what we're covering in this post.
+
+## Table of Contents
+1. [Next API Routes](#nextapiroutes)
+2. [Netlify Dev](#netlifydev)
+3. [Netlify functions with Ember](#netlifyfunctionswithember)
+4. [Netlify functions with Glimmer](#netlifyfunctionswithglimmer)
+5. [Vercel Dev](#verceldev)
+6. [Vercel functions with Ember](#vercelfunctionswithember)
+7. [Vercel functions with Glimmer](#vercelfunctionswithglimmer)
+
+## Next API Routes
+
+If you are not familiar with Next's API routes, they are essentially node functions that deploy as serverless lambdas. These functions can be used in a lot of different ways: 
+
+- they could provide a place to write a GraphQL endpoint
+- they could serve as a gateway to other API's
+- they could serve as a full-fledged backend that talks to a persistence layer
+
+There a lot of possibilities here. API routes are especially helpful when performing operations that need to access secret environment variables that you would typically not want exposed to your client. For instance, if you need to generate presigned AWS s3 urls for the client to post files to.
+
+From a developer efficiency perspective, this could help unblock frontend developers on your team from waiting for backend developers to be available for certain tasks. Since API routes provide, well, an API, you could develop certain server side features alongside your frontend without much loss of context.
+
+Developing a Next.js API route is a seamless experience. The API routes all live in an `api` directory at your project's root. Spinning up the development server that powers the client side of the Next application will also spin up a server for the api routes, do some proxy magic, and make it feel as if you have one development server.
+
+There is essentially zero-configuration involved. Vercel really nailed the developer experience with these API routes. Deploying to Vercel's platform is, of course, a snap, but you can deploy elsewhere relatively easily too.
+
+Being able to write these functions in such a well-integrated fashion was very exciting. And it left me wondering about obtaining a similar experience in the JavaScript framework that I enjoy working with the most, [Ember](https://emberjs.com/). Also having recently explored [Glimmer](https://glimmerjs.com/), I was curious about whether these functions could be used with Glimmer too. 
+
+After a bit of investigation, I'm happy to report that because of tooling that is already available it is fairly trivial to get essentially the same value-add and developer experience with both Ember and Glimmer, or any other framework you might choose.
```

Comment:
```suggestion
After a bit of investigation, I'm happy to report that, because of tooling that is already available, it is fairly trivial to get essentially the same value-add and developer experience with both Ember and Glimmer, or any other framework you might choose.
```